modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/QueryTransactionHandler.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine;
+
+import static org.apache.ignite.lang.ErrorGroups.Sql.RUNTIME_ERR;
+
+import org.apache.calcite.sql.SqlNode;
+import org.apache.ignite.internal.sql.engine.sql.IgniteSqlCommitTransaction;
+import org.apache.ignite.internal.sql.engine.sql.IgniteSqlStartTransaction;
+import org.apache.ignite.internal.sql.engine.sql.IgniteSqlStartTransactionMode;
+import org.apache.ignite.internal.sql.engine.sql.ParsedResult;
+import org.apache.ignite.internal.tx.InternalTransaction;
+import org.apache.ignite.sql.ExternalTransactionNotSupportedException;
+import org.apache.ignite.sql.SqlException;
+import org.apache.ignite.tx.IgniteTransactions;
+import org.apache.ignite.tx.TransactionOptions;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The query transaction handler is responsible for running implicit or script managed transactions during query execution.
+ */
+@SuppressWarnings("InterfaceMayBeAnnotatedFunctional")
+public interface QueryTransactionHandler {
```

Comment:
Maybe it would better have QueryTransactionHandlerImpl and ScriptTransactionHandlerImpl,
and factory methods with similar names.
As for now, methods and classes have different names and introduces to much terms...

Reply:
Done